src/main/java/com/greenbits/binlog/RowEventListenerAdapter.java

```diff
@@ -117,42 +115,43 @@ private void handleQueryEvent(QueryEvent event) {
         }
     }
 
-    private void handleXidEvent(XidEvent event) {
+    private void handleXidEvent(Event event) {
         commitTransaction(event);
     }
 
-    private void handleTableMapEvent(TableMapEvent event) {
-        tableIdToTableMapEvent.put(event.getTableId(), event);
+    private void handleTableMapEvent(Event event) {
+        TableMapEventData eventData = event.getData();
+        tableIdToTableMapEvent.put(eventData.getTableId(), eventData);
     }
 
-    private void handleRotateEvent(RotateEvent event) {
-        checkpointer.rotate(event.getBinlogFileName().toString(), event.getBinlogPosition());
+    private void handleRotateEvent(Event event) {
+        RotateEventData eventData = event.getData();
+        checkpointer.rotate(eventData.getBinlogFilename().toString(), eventData.getBinlogPosition());
     }
 
-    private void handleUnknownEvent(BinlogEventV4 event) {
-        BinlogEventV4Header header = event.getHeader();
-        throw new IllegalStateException("Unknown MySQL binlog event type " + header.getEventType() + ".");
+    private void handleUnknownEvent(Event event) {
+        throw new IllegalStateException("Unknown MySQL binlog event type " + event + ".");
     }
 
-    private void handleError(Throwable t, BinlogEventV4 event) {
+    private void handleError(Throwable t, Event event) {
         listener.onError(t, event);
     }
 
-    private void commitTransaction(AbstractBinlogEventV4 event) {
+    private void commitTransaction(Event event) {
         checkpointer.checkpoint(getNextPosition(event));
         listener.commitTransaction();
     }
 
-    private TableMapEvent getTableMapEvent(long tableId) {
-        TableMapEvent tableMapEvent = tableIdToTableMapEvent.get(tableId);
+    private TableMapEventData getTableMapEvent(long tableId) {
+        TableMapEventData tableMapEvent = tableIdToTableMapEvent.get(tableId);
         if (tableMapEvent == null) {
             throw new IllegalStateException("Cannot find table with id '" + tableId + "' in the map.");
         }
         return tableMapEvent;
     }
 
-    private long getNextPosition(BinlogEventV4 event) {
-        return event.getHeader().getNextPosition();
+    private long getNextPosition(Event event) {
+        return ((EventHeaderV4)event.getHeader()).getNextPosition();
```

Comment:
Double checking that `EventHeaderV4` is the correct class. Only asking because Open Replicator had classes with versions in the name and haven't seen anything with a version in the name until this, so just double checking.